### PR TITLE
Quiz page : Feat Options and Answers

### PR DESCRIPTION
**Pull request by shlok30**
Hi Team,

Can you please review my Quiz pages?

Clicking on an option shows you whether the option is right or wrong and score gets calculated accordingly.

Live Link : https://water-quiz-lisfodc56-shlok30.vercel.app/

Video Demo

https://user-images.githubusercontent.com/60795588/172910150-bb5618bc-4f39-45f4-8ec7-9858a47f1fc3.mov

Thanks

### CODE REVIEW

File: src/components/options.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import buttonColour from "../general-functions/buttonColour"
+import { useParams } from "react-router-dom"
+import { useQuiz } from "../context/quiz-context"
+
+const Options = ({currentQuizObject,option,index}) => {
+
+    console.log(currentQuizObject)
```
Comment (nikhil-sharma-nks) [inline src/components/options.js]: you should remove console logs when pushing to production

File: src/general-functions/calculateScore.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+
+const calculateScore = (answerList,quizID) => {
+    console.log(answerList)
```
Comment (nikhil-sharma-nks) [inline src/general-functions/calculateScore.js]: remove console log from here too

File: src/App.js [hunk 3]
```diff
@@ -1,13 +1,15 @@
 import './App.css';
 import { Route, Routes } from 'react-router-dom';
 import Homepage from './pages/homepage';
+import QuizPage from './pages/quiz-page';
```
Comment (nikhil-sharma-nks) [inline src/App.js:4]: nothing wrong with this but as your app grows you would have multiple pages and importing them becomes tedious, therefore consider making an index.js file in your pages folder where you would export all pages from there and import all pages where ever you need them in a single import line.

**Review by nikhil-sharma-nks — APPROVED** [review]
Looks good to merge. Just look into these suggestions and also use something to format your code properly.

File: src/components/options.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    const {quizID} = useParams()
+
+    const selectedAnswerObject = answerList.find(quiz => quiz.quizID === quizID)
+
+    const answersOfCurrentQuestion = [...selectedAnswerObject.answers]
+    
+    const currentAnswer = answersOfCurrentQuestion?.reduce((answer,curr) => curr.questionID === questionID ? curr.answer : answer,"")
+
+    const handleClick = () => {
+        //console.log(currentQuiz)
```
Comment (sharmayatendra) [inline src/components/options.js]: remove unnecessary code

File: src/pages/quiz-page.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    const {quizDB,questionIndex,answerList,setQuestionIndex,setAnswerList} = useQuiz()
+
+    const currentQuizObject = quizDB.find(quiz => quiz["_id"] === quizID)
+
+    const currentQuestion = currentQuizObject.mcqs[questionIndex].question
+
+    const options = currentQuizObject.mcqs[questionIndex].options
+
+    console.log("Question Index",questionIndex)
```
Comment (sharmayatendra) [inline src/pages/quiz-page.js]: remove console log

**Review by sharmayatendra — APPROVED** [review]
everything looks good 
good variable names used
you can look upon suggested changes

File: src/App.js (hunk 3 above)
Comment (shlok30) [reply src/App.js:4]: Hey this makes a lot of sense, I'll connect with you on discord to understand this better

**Pull request merged**